Fix corporatedomains whois expires_on date format

njng opened the pull request:
Fix **expires_on** for corporatedomains WHOIS parser when the date format is **MM/DD/YYYY HH:MM:SS** by changing the format before it is passed into `parse_time`. Note that there are still 2 different formats that could be possible for **expires_on**, so we only change the slash-formatted dates.

njng committed:
Fixed the WHOIS parser for corporatedomains dates using '/'

Updated the WHOIS parser for corporatedomains which were incorrectly
parsing "Registrar Registration Expiration Date", because the date had
"MM/DD/YYYY" format rather than "DD/MM/YYYY". The change updates
the format if the value has slash-separated date before passing it into
`parse_time`. This change cannot be used for all dates, since some dates
use a dash-separated format (i.e. YYYY-MM-DDTHH:MM:SSZ).

This problem also applies to the creation date as well when parsing the
value for "Creation Date".

njng committed:
Updated the specs to include testing for hyphen-separated dates

Moved the existing `status_available` files into the `com` folder where
the new `status_registered` files will be generated. Generated new
`status_registered` files for the different cases when the
creation and expiration dates are hyphen-separated vs slash-separated.

Also re-used the existing `.expected` file from the old
`status_registered` file to test for the new values.

itghlu commented on lib/whois/parsers/whois.corporatedomains.com.rb:
```diff
@@ -22,6 +22,39 @@ class WhoisCorporatedomainsCom < BaseIcannCompliant
+      def reformat_slash_date_format(value)
+        date, time = value.split
+        month, day, year = date.split("/")
+        reformatted_date = [year, month, day].join("-")
```
`Time.utc(year, month, day)` will give you a Time object which you can then convert to a string with `to_s` or `as_json`. The manual array join and string interpolation is not needed then.

njng replied:
If I'm already creating a Time object, it doesn't seem like there's any reason to convert it to a string, add the time, and then pass it in to `parse_time`, is there?

itghlu replied:
Yeah just returning the Time you generate seems good.

njng committed:
Call `Time.utc` to parse time for corporatedomains with slash-format

Return the result from `Time.utc` to parse dates using slash-format in
corporatedomains, since this would avoid using joins and string
interpolation before being passed into `parse_time`.

Renamed the method as the return value now returns a Time object rather
than a string.

itghlu commented:
LGTM
